script.js

```diff
@@ -11,29 +11,92 @@ function handleAddTodo() {
     id: Date.now(),
     title: $todoInput.value,
     createdAt: Date.now().toLocaleString(),
+    isEditing: false,
+    isCompleted: false,
   };
-  
+
   if (newTodo.title.trim() === "") return;
+
   todos.push(newTodo);
   $todoInput.value = "";
+
   renderTodos();
 }
 
 function renderTodos() {
   $todoList.innerHTML = "";
+
   todos.forEach((todo) => {
-    $todoList.append(createTodo(todo));
+    $todoList.append(createTodoItem(todo));
   });
 }
 
-function createTodo(todo) {
-  const $newTodo = document.createElement("li");
-  $newTodo.style.listStyleType = "none";
-  $newTodo.innerHTML = `<span>${todo.title}</span>`;
+function createTodoItem(todo) {
+  const $todo = document.createElement("li");
+  $todo.style.listStyleType = "none";
+
+  $todo.append(
+    todo.isEditing ? createEditTodoEl(todo) : createNormalTodoEl(todo)
+  );
+
+  $todo.append(createEditButton(todo));
+
+  return $todo;
+}
 
-  $newTodo.querySelector("span").addEventListener("click", () => {
-    $newTodo.classList.toggle("completed");
+function createNormalTodoEl(todo) {
+  const $todoText = document.createElement("span");
+  $todoText.innerText = todo.title;
+
+  $todoText.className = todo.isCompleted ? "completed" : "";
+
+  $todoText.addEventListener("click", () => {
+    toggleTodoDone(todo.id);
   });
 
-  return $newTodo;
+  return $todoText;
+}
+
+function toggleTodoDone(todoId) {
+  todos = todos.map((todo) =>
+    todo.id === todoId ? { ...todo, isCompleted: !todo.isCompleted } : todo
+  );
```

Comment:
Too much happening in one line. Let's break it down.

This'll work due to how reference work in JS
```suggestion
  const todo = todos.find(({ id }) => id === todoId)
  
  if(todo === undefined){
    return
  }
  
  todo.isCompleted = !todo.isCompleted
```